spaceflights-pyspark/hooks/post_gen_project.py

```diff
@@ -0,0 +1,30 @@
+from pathlib import Path
+
+from kedro.templates.project.hooks.utils import (
+    parse_add_ons_input,
+    setup_template_add_ons,
+    sort_requirements,
+)
+
+
+def main(selected_add_ons_list):
+    current_dir = Path.cwd()
+    requirements_file_path = current_dir / "src/requirements.txt"
+    pyproject_file_path = current_dir / "pyproject.toml"
+    python_package_name = '{{ cookiecutter.python_package }}'
+
+    # Handle template directories and requirements according to selected add-ons
+    setup_template_add_ons(selected_add_ons_list, requirements_file_path, pyproject_file_path, python_package_name)
+
+    # Sort requirements.txt file in alphabetical order
+    sort_requirements(requirements_file_path)
+
+
+if __name__ == "__main__":
+    # Get the selected add-ons from cookiecutter
+    selected_add_ons = "{{ cookiecutter.add_ons }}"
+    # Parse the add-ons to get a list
+    selected_add_ons_list = parse_add_ons_input(selected_add_ons)
+
+    if "6" in selected_add_ons_list:
+        main(selected_add_ons_list)
```

Comment:
Is this needed? My understanding was that by the time this `post_gen_project` hook gets called we already know that the user has selected `pyspark` as an option.

Reply:
This is more for when `kedro new --starter` is used which doesn't need this script to run.

Reply:
We could change the default case in `cookiecutter.json` to `all` but I think that will add some unnecessary things to the template.

Reply:
Ahhh I see. Might be good to add that as a comment in the script.

Reply:
I had a thought since none of the options of `None`, `All`, or any of the other combination of numbers don't work in this `cookiecutter.json`. Maybe we need a `null`/`as_is` option strictly for add-ons that utilise starters. This option would allow for nothing to be changed of the template. It can just be a hidden developer prompt. 

What do you guys think? @merelcht @lrcouto